common/src/wdt.rs

```diff
@@ -0,0 +1,83 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    wdt.rs
+
+Abstract:
+
+    File contains execution routines for programming the Watchdog Timer.
+
+Environment:
+
+    ROM
+
+--*/
+
+use caliptra_drivers::SocIfc;
+use core::cmp::max;
+
+use crate::cprintln;
+
+pub struct WdtTimeout(pub u32);
+
+impl Default for WdtTimeout {
+    fn default() -> WdtTimeout {
+        WdtTimeout(WDT1_TIMEOUT_SECS)
+    }
+}
+
+const EXPECTED_CALIPTRA_BOOT_TIME_IN_CYCLES: u64 = 20_000_000; // 20 million cycles
+const WDT1_TIMEOUT_SECS: u32 = 5;
+const WDT2_TIMEOUT_CYCLES: u64 = 1; // Fire immediately after WDT1 expiry
+
+/// Start the Watchdog Timer
+///
+/// # Arguments
+///
+/// * `soc_ifc` - SOC Interface
+///
+///
+pub fn start_wdt(soc_ifc: &mut SocIfc, timeout: WdtTimeout) {
+    cprintln!("[state] Starting the Watchdog Timer");
+
+    let wdt1_timeout_cycles: u64 = match soc_ifc.get_cycle_count(timeout.0) {
+        Ok(cycle_count) => max(EXPECTED_CALIPTRA_BOOT_TIME_IN_CYCLES * 10, cycle_count),
+        Err(_) => EXPECTED_CALIPTRA_BOOT_TIME_IN_CYCLES * 10,
```

Comment:
If get_cycle_count fails, should we fail this function? I assume that would mean that `timeout` is an invalid value (too large or something)?

Reply:
This is , as per Vishal , logic to assign a default value if timer period is not configured by the Soc.